### PR TITLE
OAuth documentation

### LINKED ISSUE #19: Add API usage rules page
We have yet to document things like client ratelimit, how to request a ratelimit increase, and other terms of use for the API.

Depends on:
- [x] https://github.com/osu-tournament-rating/otr-api/issues/128

### PR DESCRIPTION
**Pull request by hburn7**
Adds documentation for:

- OAuth /authorize and /refresh endpoints
- Ratelimit and terms of use
- How to use the credentials after authorizing.

Closes #19

### CODE REVIEW

File: api/usage/objects/en.md [hunk 1]
```diff
@@ -181,6 +181,16 @@ Represents a mapping of the o!TR match id to its respective osu! match id
 
 ---
 
+### OAuth Response
```
Comment (myssto) [inline api/usage/objects/en.md:184]:
```suggestion
### OAuthResponse
```
All objects use their exact class name (without DTO suffix if applicable)

File: api/usage/endpoints/oauth/en.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Example Request
+```js
+fetch("https://api.otr.stagec.xyz/api/v1/oauth/token?clientId=123&clientSecret=abcdefg", {
+  method: "POST",
+});
+```
+
+Response Format\
+Returns [OAuth Response](/api/usage/objects/en.md#oauth-response)
```
Comment (myssto) [inline api/usage/endpoints/oauth/en.md:25]:
```suggestion
Returns [OAuth Response](/api/usage/objects/en.md#oauthresponse)
```

File: api/usage/endpoints/oauth/en.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Example Request
+```js
+fetch("https://api.otr.stagec.xyz/api/v1/oauth/refresh?refreshToken=abcdefg", {
+  method: "POST",
+});
+```
+
+Response Format\
+Returns [OAuth Response](/api/usage/objects/en.md#oauth-response), containing a new `accessToken`.
```
Comment (myssto) [inline api/usage/endpoints/oauth/en.md:46]:
```suggestion
Returns [OAuth Response](/api/usage/objects/en.md#oauthresponse), containing a new `accessToken`.
```

File: api/usage/objects/en.md [hunk 4]
```diff
@@ -181,6 +181,16 @@ Represents a mapping of the o!TR match id to its respective osu! match id
...
+
+Represents a set of secrets used for Authorization. Produced after successful OAuth authentication.
+
+| Field     | Type     | Description |
+| :-------- | :------- | :---------- |
+| accessToken | string | This code is used to authorize requests. In all API requests, construct your `Authorization` header as such: `Authorization: Bearer <accessToken>`
```
Comment (myssto) [inline api/usage/objects/en.md:190]:
```suggestion
| accessToken | string | This code is used to authorize requests. In all API requests, construct your `Authorization` header as such: `Authorization: Bearer <accessToken>` |
```

File: api/usage/objects/en.md [hunk 5]
```diff
@@ -181,6 +181,16 @@ Represents a mapping of the o!TR match id to its respective osu! match id
...
+| refreshToken | string | Used by
```
Comment (myssto) [inline api/usage/objects/en.md:191]:
```suggestion
| refreshToken | string | Used to obtain a new `accessToken` |
```

File: api/usage/objects/en.md [hunk 6]
```diff
@@ -181,6 +181,16 @@ Represents a mapping of the o!TR match id to its respective osu! match id
...
 
+### OAuth Response
+
+Represents a set of secrets used for Authorization. Produced after successful OAuth authentication.
+
+| Field     | Type     | Description |
+| :-------- | :------- | :---------- |
+| accessToken | string | This code is used to authorize requests. In all API requests, construct your `Authorization` header as such: `Authorization: Bearer <accessToken>`
+| refreshToken | string | Used by 
+| accessExpiration | integer |
```
Comment (myssto) [inline api/usage/objects/en.md:192]:
```suggestion
| accessExpiration | integer | Expiration of the `accessToken` in seconds |
```

**Review by myssto — CHANGES_REQUESTED** [review]
Request again after some changes. Just small things for consistency's sake.

File: api/usage/endpoints/oauth/en.md (hunk 3 above)
Comment (hburn7) [reply api/usage/endpoints/oauth/en.md:46]: Should the link here be written as [OAuthResponse](...) based on your other comment?

File: api/usage/objects/en.md (hunk 6 above)
Comment (hburn7) [reply api/usage/objects/en.md:192]: I forgot to fill these in clearly, good catch.

File: api/usage/endpoints/oauth/en.md (hunk 3 above)
Comment (myssto) [reply api/usage/endpoints/oauth/en.md:46]: Yep, good catch

**Review by myssto — APPROVED** [review]

**Pull request merged**